packages/protocol/contracts/bridge/libs/LibBridgeData.sol

```diff
@@ -40,12 +39,13 @@ library LibBridgeData {
     event DestChainEnabled(uint256 indexed chainId, bool enabled);
 
     /**
-     * @dev Hashes messages and returns the hash signed with
-     * "TAIKO_BRIDGE_MESSAGE" for verification.
+     * @return msgHash The keccak256 hash of the message encoded with
+     * "TAIKO_BRIDGE_MESSAGE".
      */
     function hashMessage(
         IBridge.Message memory message
     ) internal pure returns (bytes32) {
+        // TODO(dave): can we use abi.encodePacked here?
         return keccak256(abi.encode("TAIKO_BRIDGE_MESSAGE", message));
```

Comment:
"a collision might happen when using keccak256 with abi.encodePacked(), but abi.encode() prevents it." -https://forum.openzeppelin.com/t/abi-encode-vs-abi-encodepacked/2948/6

I think `abi.encodePacked` may not work. If you look at the definition of `Message`:

```
struct Message {
        uint256 id; // auto filled
        address sender; // auto filled
        uint256 srcChainId; // auto filled
        uint256 destChainId;
        address owner;
        address to; // target address on destChain
        address refundAddress; // if address(0), refunds to owner
        uint256 depositValue; // value to be deposited at "to" address
        uint256 callValue; // value to be called on destChain
        uint256 processingFee; // processing fee sender is willing to pay
        uint256 gasLimit;
        bytes data; // calldata
        string memo;
    }
```

The last two properties are both dynamic size, so the follow two messages may end up with the same hash:
m1.data = bytes(string("abc"))
m1.data = string("xyz")

m2.data = bytes(string("ab"))
m2.data = string("cxyz")

Therefore, in the current code base, if there is at least one dymamic-size parameter in the encoding function, I chose to use `abi.encode`, otherwise `abi.encodePacked`.